parlai/core/worlds.py

```diff
@@ -1051,6 +1054,44 @@ def shutdown(self):
 ################################################################################
 # Functions for creating tasks/worlds given options.
 ################################################################################
+def _create_task_agents(opt):
```

Comment:
This seems fine ... i.e. we no longer have an import to a method with _ but I think it begs the question of why is this a top level function? (Same as the _path functions.) ParlAI is overall quite functional (in the programming sense). Especially here where we are building objects, why not use a builder design pattern? or factory?

Reply:
Why though? The logic could be improved, but a factory method is still a factory. It doesn't need to be wrapped as a classmethod or something.

Reply:
Fine... :) (Were there no other instances of this btw?)